### PR TITLE
backports for ActivityLog and Reporting 1.13.x

### PR DESCRIPTION
**Pull request by mpalmi**
This PR encapsulates the backports for Census and ActivityLog changes to get the release branches in sync and should resolve [VAULT-14702](https://hashicorp.atlassian.net/browse/VAULT-14702) and [VAULT-14765](https://hashicorp.atlassian.net/browse/VAULT-14765) along with the backports for the new client count testing library.

It was easier to backport the changes en masse, since they're all intertwined. Teasing apart feature-specific changes would have been more difficult.

Commits were prepared using (with fish shell, `$()` is just `()`):
```sh
git cherry-pick -v -x (git log origin/release/1.13.x..origin/main --oneline --reverse -- "*activity*" | awk '{ print $1 }')
```

and editorialized/reworded using interactive rebase. The ACME-related changes (including testing of the `clientType` proto field via the client count testing library) were not backported.

Apologies for the large PR. It may be easiest to review by comparing each backport commit with the original. I can split this up if others think it's worthwhile.

## Diff between backport and main

```sh
## NOTE: This diff does not include changes to core. 
git diff origin/activitylog-backports-1.13.x..origin/main -- "*activity*" "*census*" ':!ui/*'
```
https://gist.github.com/mpalmi/72d7f39c3d190ec660e0795419004545

[VAULT-14702]: https://hashicorp.atlassian.net/browse/VAULT-14702?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[VAULT-14765]: https://hashicorp.atlassian.net/browse/VAULT-14765?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by miagilepner** [comment]
I think you might be missing https://github.com/hashicorp/vault/pull/20626. Otherwise looks good!

File: Makefile [hunk 1]
```diff
@@ -30,16 +30,16 @@ default: dev
 bin: prep
 	@CGO_ENABLED=$(CGO_ENABLED) BUILD_TAGS='$(BUILD_TAGS) ui' sh -c "'$(CURDIR)/scripts/build.sh'"
 
+testonly:
```
Comment (miagilepner) [inline Makefile:33]: the changes in the Makefile are incorrect. The correct changes were backported.

Comment (mpalmi) [reply Makefile]: Good catch! Rebased and popped that commit out.

**Comment by raskchanky** [comment]
When I ran your `git log` command myself, I got 27 commits. There are 18 here. I'm assuming the missing 9 commits were omitted on purpose, perhaps UI related or some such?

Also, the `"*activity*"` portion of your git command makes it look like it only applies to paths with `activity` in their name somewhere. Is that only for the purpose of selecting commits? e.g. If I commit something and any of the changed files in my commit has `activity` in its name, then it shows up via this command. Is that the case? If so, were there any changes made to files w/o `activity` in their name that need to be backported as well?

**Comment by mpalmi** [comment]
> When I ran your `git log` command myself, I got 27 commits. There are 18 here. I'm assuming the missing 9 commits were omitted on purpose, perhaps UI related or some such?

While going through the backport, there were 4 classes of commits (marked below):
- ❌ empty commits, which had already been backported but weren't attributed by `git`
- ❌ superfluous changes which rejected the entire commit (these got the 'ol `--skip` during the cherry-pick)
- ✅ changes that ported over (sometimes with amendment)
- ✅ clean cherry-picks, which ported over with no modifications

It would probably be better if I had kept track of the 4 separate buckets so I could distinguish them, but for the purposes of this PR, separating by ❌ and ✅  seems sufficient.

Note: I will do something similar for the 1.11.x and 1.13.x PRs, if you think it's helpful.

----

❌  a9e17c2011 VAULT-13763 normalize activity log mount paths  (#19343)
❌  7ef729708b UI: Use specific date in clients activity test (#19419)
✅ 9f7f8d5bfa VAULT-13729 activity log test godocs (#19433)
❌  e55c18ed12 adding copyright header (#19555)
✅ e3c59773e9 Add no-op CensusAgent (#19625)
❌ e9d6dbce23 activitylog: Fix pq.Get trace logger output (#19650)
❌ bf2f1a88d6 UI: Fix flaky time-related tests (#19521)~
✅ b4fab6ac2a VAULT-13191: OSS changes (#19891)
✅ 54904e4cd6 VAULT-14733: Refactor processClientRecord in activity log (#19933)
❌  d70c17f7fd VAULT-14733: SegmentReader interface for reading activity log segments (#19934)
✅ 4b6ec4079d Require activity log retention months at least the minimum (#20078)
✅ 05ba6bbddd api: Add reporting fields to activitylog config endpoint (#20086)
✅ c95d4fb15a VAULT-14734: activity log write endpoint (#20019)
❌ 75c903f0d6 openapi: Add display attributes for /sys (p2) (#19707)
✅ 002a59a370 Add minimum_retention_months to config endpoint (#20150)
✅ 77f83d9fe8 Refactor reporter for unseal setup (#20296)
✅ 730d0e2821 VAULT-14733: Split logic of precomputedQueryWorker (#20073)
❌ d234111801 Start counting ACME certificate issuance as client activity (#20520)
✅ 35e2c1665f VAULT-15703: Reload automated reporting (#20680)
❌ 6d95f8ca07 Add client_type field to EntityRecord protobuf (#20626)
✅ 810d504e4f Add current_billing_period activity endpoint param (#20694)
✅ 5b23dd506f VAULT-14735: generate mock clients for activity log  (#20252)
✅ 018ea84997 VAULT-15395: Support mocking time functions in the activity log (#20720)
✅ 541f18eeb7 VAULT-14735: repeated and segmented activity log clients (#20699)
✅ b4e2751a09 VAULT-14735: write mock activity log entity files (#20702)
✅ dc5dd71c72 Deflake TestActivityLog_MultipleFragmentsAndSegments (#20930)
✅ 5002489d27 VAULT-15394: Generate activity log precomputed queries (#20778)

---

> Also, the `"*activity*"` portion of your git command makes it look like it only applies to paths with `activity` in their name somewhere. Is that only for the purpose of selecting commits? e.g. If I commit something and any of the changed files in my commit has `activity` in its name, then it shows up via this command. Is that the case? If so, were there any changes made to files w/o `activity` in their name that need to be backported as well?

That's right, I used it as a general heuristic for selecting the appropriate commits (with some extra post-processing). I don't think there were any files touched that did not match the `*activity*` filter, but I'll double-check -- good call!

**Review by raskchanky — APPROVED** [review]
There's a lot here to go through, especially for a backport. I did my best to sort through it, and to my eyes, it looks correct.

**Review by miagilepner — APPROVED** [review]

**Pull request merged**